src/Domain/WikiContentRendererFactory.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\ExternalContent\Domain;
+
+use ProfessionalWiki\ExternalContent\Domain\ContentRenderer\CodeRenderer;
+use ProfessionalWiki\ExternalContent\Domain\ContentRenderer\MarkdownRenderer;
+use ProfessionalWiki\ExternalContent\Domain\ContentRenderer\RendererConfig;
+
+class WikiContentRendererFactory implements ContentRendererFactory {
+
+	public function createContentRenderer( RendererConfig $config ): ContentRenderer {
+		if ( $config->language !== '' ) {
+			return new CodeRenderer(
+				language: $config->language,
+				showLineNumbers: $config->showLineNumbers
+			);
+		}
+
+		// TODO: check config if this should be default behavior
```

Comment:
TODO: Something like `$wgRenderMarkdownByDefault` (default=true), in order to keep the original behavior of the extension. However, then we need an additional parser function parameter like `render` to be used when the new config is false and you want to render Markdown instead of code.